libfwupdplugin/fu-cabinet.c

```diff
@@ -185,28 +215,64 @@ fu_cabinet_parse_release (FuCabinet *self, XbNode *release, GError **error)
 		}
 	}
 
-	/* if the signing file exists, set that too */
-	for (guint i = 0; suffixes[i] != NULL; i++) {
+	/* find out if the payload is signed, falling back to detached */
+	item = jcat_file_get_item_by_id (self->jcat_file, basename, NULL);
+	if (item != NULL) {
+		g_autoptr(GError) error_local = NULL;
+		g_autoptr(GPtrArray) results = NULL;
+		results = jcat_context_verify_item (self->jcat_context,
+						    blob, item,
+						    JCAT_VERIFY_FLAG_REQUIRE_CHECKSUM |
+						    JCAT_VERIFY_FLAG_REQUIRE_SIGNATURE,
+						    &error_local);
+		if (results == NULL) {
+			g_debug ("failed to verify payload %s: %s",
+				 basename, error_local->message);
+		} else {
+			g_debug ("verified payload %s: %u",
+				 basename, results->len);
+			release_flags |= FWUPD_RELEASE_FLAG_TRUSTED_PAYLOAD;
+		}
+
+	/* legacy GPG detached signature */
```

Comment:
I guess no extra fallback to pkcs7 is intentional, right?

Reply:
it's a good question. We only turned the PKCS-7 support on a few weeks ago on the LVFS, and so only perhaps 1% of all the firmware has both signatures. I guess we could fall back to both if you think it's worth it.

Reply:
Oh it's only been a few weeks?  Yeah I agree it's probably not worth it.  And it's probably also worth mentioning in the code comment this reason to only fall back to GPG.

Maybe also a sunset timeframe/date to remove the (legacy) code for fallback.

Reply:
>  Yeah I agree it's probably not worth it

The counterpoint here is that I'm working on a patchset that can resign the entire archive, so if we do that we'd get the pkcs "for free" even for old firmware.

> Maybe also a sunset timeframe/date to remove the (legacy) code for fallback.

How many years? Heh. RHEL.

Reply:
5?  The thing to keep in mind is metadata should always be getting refreshed.  The whole point of this stack is to be pulling in new things.  So if you're running a remote that generates old metadata formats, you're going to be forced to upgrade it at _some_ point.